src/app/select/select.directive.ts

```diff
@@ -205,4 +211,30 @@ export class MzSelectDirective extends HandlePropChanges implements AfterViewIni
 
     this.renderer.invokeElementMethod(this.selectElement, 'material_select');
   }
+
+  optionElementsLength() {
+    const length = this.selectElement[0].querySelectorAll('option').length;
+
+    if (this.placeholder) {
+      return length - 1;
+    }
+
+    return length;
+  }
+
+  listenOptionChanges() {
+    if (this.selectElement[0] && !this.optionLength) {
+      this.optionLength = this.optionElementsLength();
+    }
+
+    Observable.fromEvent($(this.selectElement), 'DOMSubtreeModified')
```

Comment:
Sadly, it doesn't work without a JQuery selector around.

`Invalid event target`.

Did I ever say that I like jQuery? :boom: